### PR TITLE
WT-11687 Create a new data structure for prefetching-related information

### PR DESCRIPTION
**Pull request by clarissecheah**
Create a new data structure for the pre-fetching variables added in the MVP.

### CODE REVIEW

File: src/btree/bt_prefetch.c [hunk 1]
```diff
@@ -72,7 +72,7 @@ __wt_btree_prefetch(WT_SESSION_IMPL *session, WT_REF *ref)
  *     reading it in is the useful side effect here. Must be called while holding a dhandle.
  */
 int
-__wt_prefetch_page_in(WT_SESSION_IMPL *session, WT_PREFETCH *pf)
+__wt_prefetch_page_in(WT_SESSION_IMPL *session, WT_PREFETCH_QUEUE_ENTRY *pf)
```
Comment (mm-ng) [inline src/btree/bt_prefetch.c:75]: Not sure if we should think about re-naming the variable from `pf` -> `pe` (short for pre-fetch entry). I'm also happy to leave it as it is.

File: src/include/btmem.h [hunk 2]
```diff
@@ -938,13 +938,23 @@ struct __wt_ref_hist {
...
 /*
  * WT_PREFETCH --
- *	Queue entry for pages queued for pre-fetch.
+ *	Pre-fetch structure containing useful information for pre-fetch.
  */
 struct __wt_prefetch {
+    WT_REF *prefetch_prev_ref;
+    uint64_t prefetch_disk_read_count; /* Sequential cache requests that caused a leaf read */
+    uint64_t prefetch_skipped_with_parent;
+};
```
Comment (mm-ng) [inline src/include/btmem.h:947]: Can we define this data structure in the `session.h` file instead? We should also delete the variables in the WT_SESSION_IMPL structure and replace it with the new data structure.

File: src/btree/bt_prefetch.c (hunk 1 above)
Comment (clarissecheah) [reply src/btree/bt_prefetch.c:75]: I think we can rename it as you suggested, might help differentiating things in the future.

File: src/include/btmem.h (hunk 2 above)
Comment (clarissecheah) [reply src/include/btmem.h:947]: Yes of course, sorry I missed adding that.

**Review by korteland — APPROVED** [review]
LGTM

**Review by mm-ng — APPROVED** [review]
LGTM

**Pull request merged**